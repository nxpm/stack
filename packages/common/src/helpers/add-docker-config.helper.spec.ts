import { Tree } from '@nrwl/devkit'
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { addDockerIgnore, addDockerfile } from './add-docker-config.helper'

describe('add-docker-config helper', () => {
  let appTree: Tree

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should add Dockerfile', async () => {
    addDockerfile(appTree)
    expect(appTree.read('Dockerfile').toString()).toMatchSnapshot()
  })
  it('should add Dockerfile', async () => {
    addDockerIgnore(appTree)
    expect(appTree.read('.dockerignore').toString()).toMatchSnapshot()
  })
})
