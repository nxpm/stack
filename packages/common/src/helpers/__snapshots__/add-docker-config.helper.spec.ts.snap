// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-docker-config helper should add Dockerfile 1`] = `
"FROM node:14-alpine

WORKDIR /workspace

COPY package.json yarn.lock /workspace/

RUN yarn

COPY . .

RUN yarn build

CMD [\\"yarn\\", \\"start\\"]"
`;

exports[`add-docker-config helper should add Dockerfile 2`] = `
"dist
node_modules"
`;
